interfaces/devdoc/srw_lock_ll_requirements.md

```diff
@@ -0,0 +1,88 @@
+# `srw_lock_ll` requirements
+
+## Overview
+
+`srw_lock_ll` is the implementation of a slim reader writer lock.
+
+## Exposed API
+
+```c
+#define SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES \
+    SRW_LOCK_LL_TRY_ACQUIRE_OK, \
+    SRW_LOCK_LL_TRY_ACQUIRE_COULD_NOT_ACQUIRE, \
+    SRW_LOCK_LL_TRY_ACQUIRE_INVALID_ARGS
+
+typedef void* SRW_LOCK_LL;
```

Comment:
>void* [](http://example.com/codeflow?start=8&length=5)

how about... instead of void... have an incomplete type here.

Such as `typedef struct HAGAUAGA* SWL_LOCK_LL". 

That'll make SRW_LOCK_LL pointers only compatible among themselves

Reply:
Changed to typedef 

#ifdef WIN32
typedef SRWLOCK SRW_LOCK_LL;
#endif